pkg/agent/host_configuration_unix.go

```diff
@@ -0,0 +1,132 @@
+		return err
+	}
+
+	// Set host gateway interface up
+	if err := netlink.LinkSetUp(link); err != nil {
+		klog.Errorf("Failed to set host link for %s up: %v", i.hostGateway, err)
+		return err
+	}
+
+	// Configure host gateway IP using the first address of node localSubnet
+	localSubnet := i.nodeConfig.PodCIDR
+	subnetID := localSubnet.IP.Mask(localSubnet.Mask)
+	gwIP := &net.IPNet{IP: ip.NextIP(subnetID), Mask: localSubnet.Mask}
+	gwAddr := &netlink.Addr{IPNet: gwIP, Label: ""}
+	gwMAC := link.Attrs().HardwareAddr
+	i.nodeConfig.GatewayConfig = &types.GatewayConfig{Link: i.hostGateway, IP: gwIP.IP, MAC: gwMAC}
+	gatewayIface.IP = gwIP.IP
+	gatewayIface.MAC = gwMAC
+
+	// Check IP address configuration on existing interface, return if already has target
+	// address
+	// We perform this check unconditionally, even if the OVS port did not exist when this
+	// function was called (i.e. portExists is false). Indeed, it may be possible for the Linux
+	// interface to exist even if the OVS bridge does not exist.
+	if addrs, err := netlink.AddrList(link, netlink.FAMILY_V4); err != nil {
+		klog.Errorf("Failed to query IPv4 address list for interface %s: %v", i.hostGateway, err)
+		return err
+	} else if addrs != nil {
+		for _, addr := range addrs {
+			klog.V(4).Infof("Found IPv4 address %s for interface %s", addr.IP.String(), i.hostGateway)
+			if addr.IP.Equal(gwAddr.IPNet.IP) {
+				klog.V(2).Infof("IPv4 address %s already assigned to interface %s", addr.IP.String(), i.hostGateway)
+				return nil
+			}
+		}
+	} else {
+		klog.V(2).Infof("Link %s has no configured IPv4 address", i.hostGateway)
+	}
+
+	klog.V(2).Infof("Adding address %v to gateway interface %s", gwAddr, i.hostGateway)
+	if err := netlink.AddrAdd(link, gwAddr); err != nil {
+		klog.Errorf("Failed to set gateway interface %s with address %v: %v", i.hostGateway, gwAddr, err)
+		return err
+	}
+
+	// send_redirects for the interface will be enabled if at least one of
+	// conf/{all,interface}/send_redirects is set to TRUE, so "all" and the
+	// interface must be disabled together.
+	// See https://www.kernel.org/doc/Documentation/networking/ip-sysctl.txt.
+	if err := disableICMPSendRedirects("all"); err != nil {
+		return err
+	}
+	if err := disableICMPSendRedirects(i.hostGateway); err != nil {
+		return err
+	}
+	return nil
+}
+
+// prepareHostNetworking returns immediately on Linux.
+func (i *Initializer) prepareHostNetworking() error {
+	return nil
+}
+
+func disableICMPSendRedirects(intfName string) error {
+	cmdStr := fmt.Sprintf("echo 0 > /proc/sys/net/ipv4/conf/%s/send_redirects", intfName)
+	cmd := exec.Command("/bin/sh", "-c", cmdStr)
+	if err := cmd.Run(); err != nil {
+		klog.Errorf("Failed to disable send_redirect for interface %s: %v", intfName, err)
+		return err
+	}
+	return nil
+}
+
+// setupExternalNetworking setups iptables chains and rules.
+func (i *Initializer) setupExternalNetworking() error {
```

Comment:
Do you think can be merged into prepareHostNetworking()?

Reply:
Ok. Do you think it can be a good idea to make SNAT by OVS a separate option if it works on Linux as well? But I guess let us keep your current implementation first.